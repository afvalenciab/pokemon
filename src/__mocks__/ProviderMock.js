import React from 'react';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../reducers';
import initialState from './initialState';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)));

const ProviderMock = (props) => {
  const { children } = props;
  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};

export default ProviderMock;
