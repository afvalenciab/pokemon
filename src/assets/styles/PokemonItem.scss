
.pokemon--item {
  display: grid;
  grid-template-rows: 80px;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 5px;
  align-items: center;
  justify-items: center;
  border-bottom: 1px solid #EDEAEA;
  cursor: pointer;

  figure {
    margin: 0;
    img {
      width: 70px;
    }
  }

  p {
    margin: 0;
  }
}

.active {
  background: rgba(0,0,0,0.4);
  border-radius: 5px;
}