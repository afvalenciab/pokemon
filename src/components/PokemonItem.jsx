import React, { useState, useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { setSelectedItems, deleteSelectedItem } from '../actions';
import '../assets/styles/PokemonItem.scss';

const PokemonItem = (props) => {
  const { name, url, setSelectedItems, deleteSelectedItem } = props;
  const [pokemon, setPokemon] = useState({
    data: {},
    className: 'pokemon--item',
  });

  useLayoutEffect(() => {
    const abortController = new AbortController();
    const { signal } = abortController;

    try {
      const fetchData = async () => {
        const response = await fetch(url, { signal });
        const data = await response.json();
        setPokemon({
          ...pokemon,
          data,
          className: 'pokemon--item',
        });
      };

      fetchData();
      return function cleanup() {
        abortController.abort();
      };
    } catch (error) {
      console.log(error);
      throw error;
    }
  }, [name]);

  const handleClickItem = () => {
    if (pokemon.className === 'pokemon--item') {
      setPokemon({
        ...pokemon,
        className: `${pokemon.className} active`,
      });
      setSelectedItems(name);
    } else {
      setPokemon({
        ...pokemon,
        className: 'pokemon--item',
      });
      deleteSelectedItem(name);
    }
  };

  return Object.keys(pokemon.data).length > 0 ? (
    <div className={pokemon.className} onClick={handleClickItem}>
      <figure className='poke--figure'>
        <img src={pokemon.data.sprites.front_default} alt={name} />
      </figure>
      <p className='poke--id'>{pokemon.data.id}</p>
      <p className='poke--name'>{name}</p>
      <p className='poke--type'>{pokemon.data.types[0].type.name}</p>
      <p className='poke--weight'>{pokemon.data.weight}</p>
    </div>
  ) : '';
};

const mapDispatchToProps = {
  setSelectedItems,
  deleteSelectedItem,
};

export default connect(null, mapDispatchToProps)(PokemonItem);
