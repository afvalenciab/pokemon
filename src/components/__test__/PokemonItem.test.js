import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import PokemonItem from '../PokemonItem';

describe('Pokemon Item Testing', () => {
  test('Match Snapshot', () => {
    const pokemon = create(
      <ProviderMock>
        <PokemonItem />
      </ProviderMock>,
    );
    expect(pokemon.toJSON()).toMatchSnapshot();
  });
});
