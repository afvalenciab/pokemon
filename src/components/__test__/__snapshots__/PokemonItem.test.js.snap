// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Item Testing Match Snapshot 1`] = `""`;
